kore/src/Kore/Builtin/Builtin.hs

```diff
@@ -983,31 +996,30 @@ getAttemptedAxiom attempt =
 
 -- | Return an unsolved unification problem.
 unifyEqualsUnsolved
-    :: (Monad m, SimplifierVariable variable)
+    :: (MonadUnify unifier, SimplifierVariable variable)
     => SimplificationType
     -> TermLike variable
     -> TermLike variable
-    -> m (Pattern variable)
-unifyEqualsUnsolved SimplificationType.And a b =
-    let
-        unified = mkAnd a b
-        predicate = makeCeilPredicate unified
-    in
-        return (pure unified) { predicate }
+    -> unifier (Pattern variable)
+unifyEqualsUnsolved SimplificationType.And a b = do
+    let unified = TermLike.markSimplified $ mkAnd a b
+    orPredicate <- Ceil.makeEvaluateTerm Predicate.topTODO unified
```

Comment:
Weren't we done with `topTODO`? Is this a new one? If memory serves, this was only supposed to be used for a limited amount of time and removed later.

Reply:
It is intended to be temporary, but I think we never decided what to do about passing predicates in during unification. So, I guess this is alright.